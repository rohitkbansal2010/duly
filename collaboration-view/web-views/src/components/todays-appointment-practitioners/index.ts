export { TodaysAppointmentPractitioners } from './todays-appointment-practitioners';
