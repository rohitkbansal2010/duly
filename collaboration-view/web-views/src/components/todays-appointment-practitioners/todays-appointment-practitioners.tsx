import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { Practitioners } from '@components/practitioners';
import { getActiveUser } from '@redux/actions';
import { RootState } from '@redux/reducers';
import { AppDispatch } from '@redux/store';
import { Practitioner } from '@types';
import { getPractitionersWithoutUser } from '@utils';

export const TodaysAppointmentPractitioners = () => {
  const practitioners = useSelector(({ TODAYS_APPOINTMENTS }: RootState) =>
    TODAYS_APPOINTMENTS.practitioners);

  const userData = useSelector(({ USER }: RootState) =>
    USER.userData);

  const dispatch: AppDispatch = useDispatch();

  useEffect(() => {
    dispatch(getActiveUser());
  }, [ dispatch ]);

  const practitionersWithoutActiveUser =
		userData && getPractitionersWithoutUser(practitioners, userData);

  const combined = userData ?
    [
      { ...userData, isCurrentUser: true },
      ...practitionersWithoutActiveUser as Practitioner[],
    ] : practitioners;

  return (
    <Practitioners
      practitioners={combined}
      isTodaysAppointment
    />
  );
};
