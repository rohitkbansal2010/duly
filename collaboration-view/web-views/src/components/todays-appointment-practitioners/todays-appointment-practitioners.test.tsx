import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { PractitionerRole } from '@enums';
import { Practitioner } from '@types';

import { TodaysAppointmentPractitioners } from './todays-appointment-practitioners';

const mockPractitioners: Practitioner[] = [ {
  id: '001',
  humanName: {
    familyName: 'qwe',
    givenNames: [ 'Foo', 'E' ],
    prefixes: [ 'Dr.' ],
  },
  role: PractitionerRole.PCP,
},
{
  id: '002',
  humanName: {
    familyName: 'asd',
    givenNames: [ 'Bar', 'E' ],
    prefixes: [ 'Dr.' ],
  },
  photo: {
    contentType: 'image/jpg',
    title: 'photo',
    size: 0,
    url: 'https://dmgwebprodstorage.blob.core.windows.net/dmgprodweb/physician-headshots/Fitzgerald_Michael_FM-003websize.jpg',
  },
  role: PractitionerRole.PCP,
},
{
  id: '003',
  humanName: {
    familyName: 'zxc',
    givenNames: [ 'Baz', 'E' ],
    prefixes: [ 'Dr.' ],
  },
  role: PractitionerRole.PCP,
} ];

const mockDispatch = jest.fn();
jest.mock('react-redux', () =>
  ({
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn().mockImplementation((selector: (state: unknown) => unknown) =>
      selector({
        TODAYS_APPOINTMENTS: { practitioners: mockPractitioners },
        USER: { userData: mockPractitioners[0] },
      })),
    useDispatch: () =>
      mockDispatch,
  }));

describe('LogoutModal', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<TodaysAppointmentPractitioners />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
