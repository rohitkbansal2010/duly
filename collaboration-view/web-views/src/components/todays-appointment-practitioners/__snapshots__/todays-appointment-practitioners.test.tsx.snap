// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutModal should match snapshot 1`] = `
<Practitioners
  isTodaysAppointment={true}
  practitioners={
    Array [
      Object {
        "humanName": Object {
          "familyName": "qwe",
          "givenNames": Array [
            "Foo",
            "E",
          ],
          "prefixes": Array [
            "Dr.",
          ],
        },
        "id": "001",
        "isCurrentUser": true,
        "role": "PCP",
      },
      Object {
        "humanName": Object {
          "familyName": "asd",
          "givenNames": Array [
            "Bar",
            "E",
          ],
          "prefixes": Array [
            "Dr.",
          ],
        },
        "id": "002",
        "photo": Object {
          "contentType": "image/jpg",
          "size": 0,
          "title": "photo",
          "url": "https://dmgwebprodstorage.blob.core.windows.net/dmgprodweb/physician-headshots/Fitzgerald_Michael_FM-003websize.jpg",
        },
        "role": "PCP",
      },
      Object {
        "humanName": Object {
          "familyName": "zxc",
          "givenNames": Array [
            "Baz",
            "E",
          ],
          "prefixes": Array [
            "Dr.",
          ],
        },
        "id": "003",
        "role": "PCP",
      },
    ]
  }
/>
`;
